spec/classes/foreman_proxy__proxydhcp__spec.rb

```diff
@@ -194,6 +194,60 @@
             'interfaces' => ['eth0']
         ) end
       end
+
+      context "as manager of ACLs for dhcp" do
+        let :facts do
+          facts.merge({:ipaddress_eth0 => '192.168.100.21',
+                       :ipaddress      => '192.168.100.21',
+                       :netmask_eth0   => '255.255.255.0',
+                       :network_eth0   => '192.168.100.0',
+                       :osfamily       => 'RedHat'})
+        end
+
+        let :pre_condition do
+          "class {'foreman_proxy':
+            dhcp_range        => '192.168.100.1 192.168.100.10',
+            dhcp_node_type    => 'secondary',
+            dhcp_peer_address => '192.168.1.20',
+            dhcp_manage_acls  => true,
+          }"
+        end
+
+        it do should contain_exec('setfacl_etc_dhcp').
+          with_command("setfacl -R -m u:foreman-proxy:rx /etc/dhcp")
+        end
+
+        it do should contain_exec('setfacl_var_lib_dhcp').
+          with_command("setfacl -R -m u:foreman-proxy:rx /var/lib/dhcpd")
+        end
+      end
+
+      context "does not manage dhcp ACLs on non-RedHat system" do
```

Comment:
This test needs updating.

Reply:
What part needs updating?

Reply:
For Debian it also sets the exec if you explicitly tell it to so that makes it fail.

Another way of testing it is adding the following in the default params test:
```ruby
if facts[:osfamily] == 'RedHat' do
  it { should contain_exec() }
else
  it { should_not contain_exec() }
end
```

This saves (run)time and is also clearer because right now you have the following output:

```
on redhat-7-x86_64
  does not manage dhcp ACLs on non-RedHat system
```

You can consider ```context "does not manage", if: facts[:osfamily] == 'Debian'``` but I think my first suggestion provides the same benefit.